ghost/recommendations/src/RecommendationService.ts

```diff
@@ -142,17 +142,23 @@ export class RecommendationService {
     }
 
     async checkRecommendation(url: URL): Promise<Partial<RecommendationPlain>> {
-        // If a recommendation with this URL already exists, return it, but with updated metadata
-        const existing = await this.repository.getByUrl(url);
+        let metadata;
+        try {
+            metadata = await this.recommendationMetadataService.fetch(url);
+        } catch (e) {
+            metadata = {};
+        }
+
+        const updatedUrl = metadata.url || url;
+        const existing = await this.repository.getByUrl(updatedUrl);
+
+        // If a recommendation with this URL already exists, return it
         if (existing) {
-            this._updateRecommendationMetadata(existing);
-            await this.repository.save(existing);
             return existing.plain;
```

Comment:
This no longer updates the existing recommendation in the database, why is that?